Code/GraphMol/DistGeomHelpers/Embedder.cpp

```diff
@@ -922,37 +923,37 @@ bool setupInitialBoundsMatrix(
 }  // namespace EmbeddingOps
 
 void _fillAtomPositions(RDGeom::Point3DConstPtrVect &pts, const Conformer &conf,
-                        const ROMol &mol, bool onlyHeavyAtomsForRMS) {
-  unsigned int na = conf.getNumAtoms();
-  pts.clear();
-  pts.reserve(na);
-  for (const auto &atom : mol.atoms()) {
-    // FIX: should we include D and T here?
-    if (onlyHeavyAtomsForRMS && atom->getAtomicNum() == 1) {
-      continue;
-    }
-    pts.push_back(&conf.getAtomPos(atom->getIdx()));
+                        const ROMol &mol,
+                        const std::vector<unsigned int> &match) {
+  PRECONDITION(pts.size() == match.size(), "bad pts size");
+  for (unsigned int i = 0; i < match.size(); i++) {
+    pts[i] = &conf.getAtomPos(match[i]);
   }
 }
 
-bool _isConfFarFromRest(const ROMol &mol, const Conformer &conf,
-                        double threshold, bool onlyHeavyAtomsForRMS) {
+bool _isConfFarFromRest(
+    const ROMol &mol, const Conformer &conf, double threshold,
+    const std::vector<std::vector<unsigned int>> &selfMatches) {
   // NOTE: it is tempting to use some triangle inequality to prune
   // conformations here but some basic testing has shown very
   // little advantage and given that the time for pruning fades in
   // comparison to embedding - we will use a simple for loop below
   // over all conformation until we find a match
-  RDGeom::Point3DConstPtrVect refPoints, prbPoints;
-  _fillAtomPositions(refPoints, conf, mol, onlyHeavyAtomsForRMS);
+  RDGeom::Point3DConstPtrVect refPoints(selfMatches[0].size());
+  RDGeom::Point3DConstPtrVect prbPoints(selfMatches[0].size());
+  _fillAtomPositions(refPoints, conf, mol, selfMatches[0]);
 
   double ssrThres = conf.getNumAtoms() * threshold * threshold;
-  for (auto confi = mol.beginConformers(); confi != mol.endConformers();
-       ++confi) {
-    _fillAtomPositions(prbPoints, *(*confi), mol, onlyHeavyAtomsForRMS);
-    RDGeom::Transform3D trans;
-    auto ssr = RDNumeric::Alignments::AlignPoints(refPoints, prbPoints, trans);
-    if (ssr < ssrThres) {
-      return false;
+  for (const auto &match : selfMatches) {
+    for (auto confi = mol.beginConformers(); confi != mol.endConformers();
```

Comment:
Should you loop through confs -> matches  or matches -> confs?

Reply:
I don't think it matters? Am I missing something?

Reply:
The self matches are contiguous in memory and will most likely be faster to access in the inner loop.

Reply:
Ah, makes sense.
Given the call to the alignment code that's in the loop, I think this is unlikely to be a bottleneck here. I can change it if you think it'll make a difference

Reply:
It makes my premature optimization sensor happy.